proposals/0282-record-dot-syntax.rst

```diff
@@ -68,15 +69,37 @@ For the specification we focus on the changes to the parsing rules, and
 the desugaring, with the belief the type checking and renamer changes
 required are an unambiguous consequences of those.
 
-2.1 ``RecordDotSyntax`` language extension
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+2.1 Language extensions
+~~~~~~~~~~~~~~~~~~~~~~~
 
-This change adds a new language extension ``RecordDotSyntax``.
+This change adds new language extensions ``OverloadedRecordDot`` and
+``OverloadedRecordUpdate``.
+
+If ``OverloadedRecordDot`` is on:
+
+- The expression ``.lbl`` means ``getField @"lbl"``;
+- The expression ``e.lbl`` means ``getField @"lbl" e``.
+
+If ``OverloadedRecordDot`` is not on, these expressions are parsed as
+uses of the function ``(.)``.
+
+If ``OverloadedRecordUpdate`` is on:
+
+- The expression ``e{lbl = val}`` means ``setField @"lbl" val``;
+- Update expressions with qualified labels like ``r{M.x = val}`` are disallowed.
+
+If ``OverloadedRecordUpdate`` is not on, ``e{lbl = val}`` means just
+what it does in Haskell98.
+
+If ``OverloadedRecordDot`` and ``OverloadedRecordUpdate`` are on:
+
+- The expression ``e{lbl₁.lbl₂ = val}`` means ``e{lbl₁ = (e.lbl₁){lbl2 = val}}``.
 
```

Comment:
Add "Otherwise the expression ``e {lbl1.lbl2 = val}`` is illegal".

Reply:
- Done